frontend/src/services/api.js

```diff
@@ -0,0 +1,224 @@
+  {
+    id: 1,
+    brand: "cerave",
+    name: "hydrating cleanser",
+    ingredient_list: ["water", "glycerin", "cocamidopropyl betaine", "sodium lauroyl sarcosinate", "peg-150 pentaerythrityl tetrastearate", "niacinamide", "peg-6 caprylic/capric glycerides", "sodium hyaluronate", "ceramide np", "ceramide ap", "ceramide eop", "carbomer", "methylparaben", "sodium chloride", "sodium lauroyl lactylate", "cholesterol", "phenoxyethanol", "disodium edta", "dipotassium phosphate", "sodium phosphate", "tocopherol", "phytosphingosine", "xanthan gum", "ethylhexylglycerin"]
+  },
+  {
+    id: 2,
+    brand: "the ordinary",
+    name: "niacinamide 10% + zinc 1%",
+    ingredient_list: ["aqua", "niacinamide", "pentylene glycol", "zinc pca", "dimethyl isosorbide", "tamarindus indica seed gum", "xanthan gum", "isoceteth-20", "ethoxydiglycol", "phenoxyethanol", "chlorphenesin"]
+  },
+  {
+    id: 3,
+    brand: "neutrogena",
+    name: "hydro boost water gel",
+    ingredient_list: ["water", "dimethicone", "glycerin", "cetearyl olivate", "sorbitan olivate", "phenoxyethanol", "synthetic beeswax", "trehalose", "carbomer", "sodium hyaluronate", "ethylhexylglycerin", "c12-20 alkyl glucoside", "sodium hydroxide", "caprylyl glycol"]
+  },
+  {
+    id: 4,
+    brand: "la roche posay",
+    name: "toleriane caring wash",
+    ingredient_list: ["aqua", "glycerin", "sodium cocoyl isethionate", "arginine", "peg-200 hydrogenated glyceryl palmate", "coco-betaine", "peg-7 glyceryl cocoate", "phenoxyethanol", "peg-120 methyl glucose dioleate", "niacinamide", "sodium chloride", "sodium benzoate", "citric acid", "zinc gluconate", "copper gluconate", "manganese gluconate"]
+  },
+  {
+    id: 5,
+    brand: "olay",
+    name: "regenerist micro-sculpting serum",
+    ingredient_list: ["water", "dimethicone", "isohexadecane", "glycerin", "isopropyl isostearate", "polyacrylamide", "olea europaea fruit oil", "tocopheryl acetate", "sodium pca", "c13-14 isoparaffin", "dimethiconol", "laureth-7", "cyclopentasiloxane", "phenoxyethanol", "methylparaben", "ethylparaben", "propylparaben", "butylparaben", "isobutylparaben", "fragrance"]
+  },
+  {
+    id: 6,
+    brand: "aveeno",
+    name: "daily moisturizing lotion",
+    ingredient_list: ["water", "glycerin", "distearyldimonium chloride", "petrolatum", "isopropyl palmitate", "cetyl alcohol", "avena sativa kernel flour", "benzyl alcohol", "sodium chloride"]
+  }
+]
+
+async function getAllProducts(maxProducts = 20, pageNumber = 1) {
+  try {
+    const requestUrl = `${API_WEBSITE}/products?limit=${maxProducts}&page=${pageNumber}`
+
+    const response = await fetch(requestUrl, {
+      method: 'GET', 
+      headers: {
+        'Accept': 'application/json', 
+      },
+    })
+
+    if (!response.ok) {
+      throw new Error(`API request failed! Status: ${response.status}`)
+    }
+
+    return await response.json()
+
+  } catch (error) {
+    console.warn('Main API not working, using backup data:', error.message)
```

Comment:
can you create a custom error handler function that takes care of logging and handling errors in a centralized way or return a fallback response to the user, indicating that the API is not available.